### DISCUSSION TITLE
Reverse-proxy nginx trouble

### DISCUSSION THREAD

**Discussion by snap87** [infinispan/infinispan#12675] [General]
When using the nginx reverse proxy with the infinispan console (v.13 or v.14), an error like this constantly occurs:
`2024/07/22 12:45:26 [error] 28529#28529: *134318 upstream sent "Content-Length" and "Transfer-Encoding" headers at the same time while reading response header from upstream, client: 10.0.1.179, server: infinispan-dev, request: "GET /console
/fonts/RedHatText-Regular.woff2 HTTP/1.1", upstream: "http://10.0.0.1:11222/console/fonts/RedHatText-Regular.woff2", host: "infinispan*", referrer: "https://infinispan*/console/app.css`

current nginx setting is as follows:

````
upstream infinispan {    
     server 10.0.1.194:11222;    
     server 10.0.1.195:11222;    
     server 10.0.1.196:11222;    
}



server {
    listen                  443 ssl;
    server_name             infinispan*;
    ssl_certificate	    /etc/nginx/certs/fullchain1.pem;
    ssl_certificate_key     /etc/nginx/certs/privkey1.pem;
    client_max_body_size    32m;
    error_page              500 502 503 504  /50x.html;
    location / {
       proxy_pass http://infinispan;
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_connect_timeout 300;
      # Default is HTTP/1, keepalive is only enabled in HTTP/1.1
      proxy_http_version 1.1;
      proxy_set_header Connection "";
      chunked_transfer_encoding off;

    }
    location = /50x.html {
          root              /var/lib/nginx/html;
    }
}

Can you suggest a correct setting for working with the console?

**Comment by jabolina** [comment]
Hey, @snap87, thanks for reporting. Does this happen on a specific page or just by opening the console? RFC 9112 doesn't allow those headers together. I tried finding a response violating that but couldn't identify any on the pages I looked at.

**Reply by snap87** [reply]
This happens when I try to open the console page. It opens very poorly with a bunch of 502 errors. Without a reverse proxy it opens without problem

**Reply by jabolina** [reply]
I see it now. It was fixed on 15.0.0+ by [ISPN-14345](https://issues.redhat.com/browse/ISPN-14345). However, on the latest 14.0, I can see the issue. I don't think we'll have a new 14 (or 13) release soon, but I created [ISPN-16321](https://issues.redhat.com/browse/ISPN-16321) for this.

On the NGINX side, I am not very familiar with its configuration to know if there is a workaround for disabling the validation of headers.

**Reply by snap87** [reply]
I see, thank you. I hope that in the 13th and 14th they will release a small patch to fix this.

**Comment by snap87** [comment]
The problem was observed in versions 13 and 14 due to a bug and was fixed in version 15.
